forge-gui/res/cardsfolder/upcoming/invasion_of_tarkir_defiant_thundermaw.txt

```diff
@@ -2,9 +2,10 @@ Name:Invasion of Tarkir
 ManaCost:1 R
 Types:Battle Siege
 Defense:5
-T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigReveal | TriggerDescription$ When CARDNAME enters the battlefield, reveal any number of Dragon cards from your hand. When you do, CARDNAME deals X plus 2 damage to any other target, where X is the number of cards revealed this way. (X can be 0.)
-SVar:TrigReveal:DB$ Reveal | RevealValid$ Card.Dragon+YouCtrl | AnyNumber$ True | RememberRevealed$ True | SubAbility$ DBDamage
-SVar:DBDamage:DB$ DealDamage | ValidTgts$ Creature,Player,Planeswalker,Battle | TgtPrompt$ Select any target | NumDmg$ X | SubAbility$ DBCleanup
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigReveal | TriggerDescription$ When CARDNAME enters the battlefield, reveal any number of Dragon cards from your hand. 
+SVar:TrigReveal:DB$ Reveal | RevealValid$ Card.Dragon+YouCtrl | AnyNumber$ True | RememberRevealed$ True | SubAbility$ DBImmediateTrig
+SVar:DBImmediateTrig:DB$ ImmediateTrigger | ConditionDefined$ Remembered | ConditionPresent$ GE1 | Execute$ TrigDamageBis | TriggerDescription$ When you do, CARDNAME deals X plus 2 damage to any other target, where X is the number of cards revealed this way. (X can be 0.)
+SVar:TrigDamageBis:DB$ DealDamage | ValidTgts$ Creature.Other,Player,Planeswalker.Other,Battle.Other | TgtPrompt$ Select any target | NumDmg$ X | SubAbility$ DBCleanup
```

Comment:
You need to remember it on the trigger

Reply:
TgtPrompt$ Select any other target

Reply:
It doesn't work because you cleanup while resolving the first trigger
so how can the second get the info it needs?
That's why I said:
"You need to remember it on the trigger"

Reply:
You also need to use `X;TriggerRemembered$...`

Reply:
@tool4ever I don't really follow anymore where the cleanup needs to happen. This version seems to work during testing ?

Does the cleanup need to happen between the first and immediatetrigger ?

Reply:
Sure it works but if I _Stifle_ the reflexive trigger your version will keep the Remembered (which arguably is "just" an annoyance here since it's ETB but easily lead to bug reports in the past because of unforeseen results on other cards)

That's why when you remember you always clean it up asap

Tbh it's pretty much like you would script an effect, just slightly different syntax:
1. You remember something
2. You create an effect (trigger)
3. If the effect (trigger) needs to use some of the remembered stuff you pass it over with RememberObjects$
4. now you clean up

Reply:
Oh thanks for the explanation ! 

Since this has been going back and forth for a while I just want to confirm that we need a SubAbility$ Cleanup on line 7 ? I though this was deemed incorrect in the past but that might be my bad

Reply:
Yes just move it, that makes all the difference

Just like an Effect the Trigger will just disappear and not need its own Cleanup by default (unless you would start Remembering something as part of it again)

Reply:
Ok Thanks ! must have misunderstood in the past